yetanotherco/aligned_layer
# fix(telemetry): remove sensitive data exposure

# Telemetry: remove sensitive data exposure

## Description

Changes the `/versions` endpoint to only expose the following fields of the operator:

- operator_id
- name
- version
- address
- stake

## How to Test

1. Start anvil:

```bash
make anvil_start_with_block_time
```

2. Start telemetry:

```bash
make telemetry_clean_db
make telemetry_full_start
```

3. Start aggregator:

```bash
make aggregator_start
```

4. Register and start an operator:

```bash
make operator_register_and_start
```

5. Go to `localhost:4001/versions` and check that the displayed data is the expected.

6. Also check the endpoint for that specific operator in `localhost:4001/versions/{operator_address}`.

7. Finally, check that `localhost:4001/api/operators` and `localhost:4001/api/operators/{operator_address}` keeps displaying all operator data, not just the public.


> [!NOTE]
> The version may appear with a `null` value. This is because the `version` variable in the operator is set in compile time and we are not running the operator compiled binary.


## Type of change


## Checklist

  - [Link to the open issue addressing it]()
